### ISSUE TITLE
Nil dereference in bearer.go

### ISSUE THREAD

**Issue by drigz** [cloudprober/cloudprober#102]
**Describe the bug**
I observed the following segfault, I'm not 100% sure why:

```
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                            
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd198b5]                                                                                                                                             
                                                                                                                                                                                                                   
goroutine 1 [running]:                                                                                                                                                                                             
github.com/cloudprober/cloudprober/common/oauth.glob..func3(0xc0001f97c0)                                                                                                                                          
    /home/runner/work/cloudprober/cloudprober/common/oauth/bearer.go:63 +0x115                                                                                                                                     
github.com/cloudprober/cloudprober/common/oauth.newBearerTokenSource(0xc0005ad840, 0xc0006e74d0)                                                                                                                   
    /home/runner/work/cloudprober/cloudprober/common/oauth/bearer.go:86 +0x15c                                                                                                                                     
github.com/cloudprober/cloudprober/common/oauth.TokenSourceFromConfig(0x19fd180, 0x0)                                                                                                                              
    /home/runner/work/cloudprober/cloudprober/common/oauth/oauth.go:36 +0x4c6                                                                                                                                      
github.com/cloudprober/cloudprober/probes/http.(*Probe).Init(0xc0002eec40, {0xc0005c8311, 0x0}, 0x0)                                                                                                               
    /home/runner/work/cloudprober/cloudprober/probes/http/http.go:223 +0x745                                                                                                                                       
github.com/cloudprober/cloudprober/probes.initProbe(0xc000227100, 0xc000194cc0)                                                                                                                                    
    /home/runner/work/cloudprober/cloudprober/probes/probes.go:186 +0x763                                                                                                                                          
github.com/cloudprober/cloudprober/probes.CreateProbe(0xc000227100, 0xc000194cc0)                                                                                                                                  
    /home/runner/work/cloudprober/cloudprober/probes/probes.go:113 +0x3b                                                                                                                                           
github.com/cloudprober/cloudprober/prober.(*Prober).addProbe(0xc00081d290, 0xc000227100)                                                                                                                           
    /home/runner/work/cloudprober/cloudprober/prober/prober.go:112 +0x35b                                                                                                                                          
github.com/cloudprober/cloudprober/prober.(*Prober).Init(0xc00081d290, {0x19d2050, 0xc00033ef40}, 0xc0002ee1c0, 0x0)                                                                                               
    /home/runner/work/cloudprober/cloudprober/prober/prober.go:186 +0x47d                                                                                                                                          
github.com/cloudprober/cloudprober.InitFromConfig({0xc00013c000, 0x167})                                                                                                                                           
    /home/runner/work/cloudprober/cloudprober/cloudprober.go:206 +0x5e9                                                                                                                                            
main.main()                                                                                                                                                                                                        
    /home/runner/work/cloudprober/cloudprober/cmd/cloudprober.go:180 +0x590
```

**Cloudprober Version**
v0.11.5

**To Reproduce**

1. Enable GKE workload identity.
2. Run Cloudprober on GKE.
3. Create a probe with a bad Service Account name (there may be another way to reproduce this, IIRC I saw it with "default" initially but can't reproduce that now)

```
    probe {
      name: "test"
      type: HTTP
      targets {
        host_names: "www.google.com"
      }
      http_probe {
        protocol: HTTPS
        relative_url: "/"
        oauth_config {
          bearer_token {
            gce_service_account: "notarealsa"
          }
        }
      }
    }
```

**Additional context**

It seems the error that causes the crash is suppressed by the crash. I was able to get it out by checking the logs of the gke-metadata-server pod running on the same node.

**Comment by manugarg** [comment]
Ah, I see the root cause:

https://github.com/cloudprober/cloudprober/blob/81b0b5a5837b75b158a46d404b864df025ed34d7/common/oauth/bearer.go#L63

So every time there is an error in retrieving token from GCE compute, it will panic.

Thanks for reporting this. Submitting the fix for this shortly.

Since I just released v0.11.5, I'll cut a new release with the bug fix and combine release notes.

**Comment by manugarg** [comment]
Cut a new release with the fix: [v0.11.6](https://github.com/cloudprober/cloudprober/releases/tag/v0.11.6).

Thanks a lot for reporting the bug, @drigz.
